File: 2279. Maximum Bags With Full Capacity of Rocks/Approach 1 - Greedy/Maximum Bags With Full Capacity of Rocks.py
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
        remaining_capacity = [cap - rock for cap, rock in zip(capacity, rocks)]
        remaining_capacity.sort()
        full_bags = 0
        
        for curr_capacity in remaining_capacity:
            if additionalRocks >= curr_capacity:
                additionalRocks -= curr_capacity
                full_bags += 1
            else:
                break
        
        return full_bags

File: Problems & Solutions/2279. Maximum Bags With Full Capacity of Rocks/27 December 2022/Approach 1 - Greedy/Maximum Bags With Full Capacity of Rocks.cpp
class Solution {
public:
    int maximumBags(vector<int>& capacity, vector<int>& rocks, int additionalRocks) {
        int n = int(capacity.size()), fullBags = 0;

        vector<int> remainingCapacity(n);

        for (int i = 0; i < n; ++i)
            remainingCapacity[i] = capacity[i] - rocks[i];
        sort(remainingCapacity.begin(), remainingCapacity.end());

        for (int i = 0; i < n; ++i) {
            if (additionalRocks >= remainingCapacity[i]) {
                additionalRocks -= remainingCapacity[i];
                fullBags++;
            }
            else {
                break;
            }
        }
        
        return fullBags;
    }
};